library/core/src/slice/cmp.rs

```diff
@@ -92,20 +91,28 @@ where
 }
 
 // Use memcmp for bytewise equality when the types allow
-impl<A, B> SlicePartialEq<B> for [A]
-where
-    A: BytewiseEq<B>,
-{
-    fn equal(&self, other: &[B]) -> bool {
-        if self.len() != other.len() {
-            return false;
-        }
+// and `memcmp` is not equivalent to our generic case
+#[cfg(not(target_env = "musl"))]
```

Comment:
that cfg needs a clear comment (saying that musl's memcmp is not more optimized)

Reply:
Thx for reviewing.
I'm planning on commenting more here and in `pal.rs` but I'm waiting to see how the debate on cfg in `core` goes.